drivers/devicetree/Make.defs

```diff
@@ -0,0 +1,28 @@
+############################################################################
+# drivers/devicetree/Make.defs
```

Comment:
do you plan to implement the full feature of device tree? it's better to move into drivers/misc if not.

Reply:
Depends on what you mean by that. I plan to implement node and cell traversal / parsing which will then let me use this in virtio and possibly some other drivers.

Reply:
> Depends on what you mean by that. I plan to implement node and cell traversal / parsing which will then let me use this in virtio and possibly some other drivers.

It's good idea to initialize drivers(at least virtio) from device tree. As I understand, we can just expose some fdt api:

1. either parse the fdt return the node with the specific compatible string.
2. or call the user supply callback in each node

And let's the board files or virtio subsystem get the virtio device info from this api and populate the related drivers.

BTW, @CV-Bowen is working to improve virtio driver framework initialized by @masayuki2009 :

1. A general framework to support any virtio driver
2. Implement the core drivers(rng, serial, net, vsock)
3. Reuse OpenAMP infrastructure

The change will upstream in the next couple weaks. Let's align our effort.
Before Q3, more drivers will be added(e.g. wifi, bt, gpu, touch, sound, camera).

Reply:
This sounds great, I can first wire the existing driver to be able to use what I am doing then.  My thought was to have the board iterate over the nodes and use that data to instantiate the virtio devices.  The procfs interface here is just for being able to write a debug tool dump the tree in user readable (src) form.  We could expose it hierarchically in procfs like `/sys/firmware/devicetree/*` but probably wont do that unless there is a good reason to at this point.